### PR TITLE
Batch http headers

### PR DESCRIPTION
**Pull request by nsarris**
@watfordgnf I found a way to test the batch headers by intercepting the http request through settings. This is a version without any functional changes. Some whitespace changes here are there are from VS auto formatting. Otherwise its pretty lean and straight forward.

### CODE REVIEW

**Review by object — CHANGES_REQUESTED** [review]
I have just a code styling comment. The class BatchWriteBase has several member variables (_session, _ entryMap etc.) and all of them are currently using underscode prefix convention. If you extend the class with a new member variable, it's better to keep consistent naming convention rather than using a new one. So I recommend changing the name to _headers to keep it consistent with the rest of the member variables in this class.

**Comment by nsarris** [comment]
@object I think you meant the ODataBatch class, that's the only one I added a headers field to. Renamed to match convention.

File: src/Simple.OData.Client.Core/ODataBatch.cs [hunk 1]
```diff
@@ -14,8 +14,8 @@ public class ODataBatch
     {
         private readonly ODataClient _client;
         private readonly List<Func<IODataClient, Task>> _actions = new List<Func<IODataClient, Task>>();
-        private readonly ConcurrentDictionary<object, IDictionary<string, object>> _entryMap = new ConcurrentDictionary<object, IDictionary<string, object>>(); 
-
+        private readonly ConcurrentDictionary<object, IDictionary<string, object>> _entryMap = new ConcurrentDictionary<object, IDictionary<string, object>>();
+        private readonly Dictionary<string, string> _headers = new Dictionary<string, string>();
         /// <summary>
```
Comment (watfordgnf) [inline src/Simple.OData.Client.Core/ODataBatch.cs:19]:
```suggestion
        private readonly Dictionary<string, string> _headers = new Dictionary<string, string>();
        
        /// <summary>
```

File: src/Simple.OData.Client.Core/Adapter/BatchWriterBase.cs [hunk 2]
```diff
@@ -47,6 +48,10 @@ protected BatchWriterBase(ISession session, IDictionary<object, IDictionary<stri
             {
                 // Create batch request message
                 var requestMessage = await EndBatchAsync().ConfigureAwait(false);
+
+                foreach (var header in headers)
+                    requestMessage.Headers.Add(header.Key, header.Value);
```
Comment (watfordgnf) [inline src/Simple.OData.Client.Core/Adapter/BatchWriterBase.cs:53]:
```suggestion
                foreach (var header in headers)
                {
                    requestMessage.Headers.Add(header.Key, header.Value);
                }
```

File: src/Simple.OData.Client.Core/ODataBatch.cs [hunk 3]
```diff
@@ -81,7 +81,33 @@ public Task ExecuteAsync()
...
+        /// <summary>
+        /// Adds a collection of headers to be included in the HTTP request.
+        /// </summary>
+        /// <param name="name">The header name.</param>
+        /// <param name="value">The header value.</param>
+        /// <returns>Self.</returns>
+        public ODataBatch WithHeaders(IDictionary<string,string> headers)
+        {
+            foreach (var header in headers)
+                WithHeader(header.Key, header.Value);
```
Comment (watfordgnf) [inline src/Simple.OData.Client.Core/ODataBatch.cs:108]:
```suggestion
            foreach (var header in headers)
            {
                WithHeader(header.Key, header.Value);
            }
```

**Comment by nsarris** [comment]
@watfordgnf all changes commited.

**Pull request merged**